src/components/item/ItemsContainer.js

```diff
@@ -8,133 +8,30 @@ import Item from 'components/item/Item';
-          currentCart.forEach(cart => {
-              if (parseInt(item) === cart.product_id) {
-                  itemToAdd.customerSize = { ...cart.customerSize };
-                  alreadyInCart = true;
-              }
-          });
-          // If not in the cart initalize new selected size and quantity object
-          if (alreadyInCart === false) {
-              itemToAdd.customerSize = { s: 0, m: 0, l: 0 };
-          }
-
-          // Check what size selected and see if;
-          // Item is in stock or
-          // Item has the desired amount in stock 
-          if (size.size === 's') {
-              checkSizeOfItem(itemToAdd, parseInt(numberOfItems.number), 's', alreadyInCart);
-          } else if (size.size === 'm') {
-              checkSizeOfItem(itemToAdd, parseInt(numberOfItems.number), 'm', alreadyInCart);
-          } else if (size.size === 'l') {
-              checkSizeOfItem(itemToAdd, parseInt(numberOfItems.number), 'l', alreadyInCart);
-          }
-          // If item is already in the cart don't add it again
-          if (alreadyInCart === true) {
-              dispatch({ type: 'setCurrentSize', payload: { size: null, id: null } });
-              return;
-          }
-
-          currentCart.push(itemToAdd);
-          dispatch({ type: 'addItemToCart', payload: currentCart });
-          dispatch({ type: 'setCurrentSize', payload: { size: null, id: null } });
-      }
-    };
-
-    const checkSizeOfItem = (itemBeingAdded, numberOfItems, sizeOfItem, isInCart) => {
-      if (itemBeingAdded.size[sizeOfItem] === 0) {
-          alert(`This item is out of stock`);
-          return;
-      } else if (itemBeingAdded.size[sizeOfItem] < numberOfItems) {
-          alert(`Only added ${itemBeingAdded.size[sizeOfItem]} items. Don't have ${numberOfItems} in stock`);
-          numberOfItems = itemBeingAdded.size[sizeOfItem];
-      }
-
-      const updatedProducts = products.map(product => {
-          if (product.product_id === itemBeingAdded.product_id) {
-              product.size[sizeOfItem] -= numberOfItems;
-          }
-          return product;
-      });
-      
-      dispatch({ type: 'updateProducts', payload: updatedProducts });
-
-      isInCart ?
-          itemBeingAdded.customerSize = { ...itemBeingAdded.customerSize, [sizeOfItem]: (numberOfItems += parseInt(itemBeingAdded.customerSize[sizeOfItem])) } :
-          itemBeingAdded.customerSize = { ...itemBeingAdded.customerSize, [sizeOfItem]: numberOfItems }
-
-      return itemBeingAdded;
-    };
-
-
     let itemsList = [];
     if (!hasBeenFiltered) {
       itemsList = products.map((product) => (
         <Item
-          key={product.product_id}
-          addToCart={addToCart}
+          key={product.id}
           product={product} />
       ));
     } else if (hasBeenFiltered && currentProducts.length) {
       itemsList = currentProducts.map((product) => (
         <Item
-          key={product.product_id}
```

Comment:
The naming is redundant...IMO it should be `product.skuid` which is a typical way for sales-y things to identify products. In your implementation the user cannot add the same product to their cart twice (dunno why 🤷 ) but you can do this in amazon etc., so you need one id for the product and another id for the basket item